chapters/annotations.tex

```diff
@@ -2195,10 +2203,10 @@ \subsection{Licensing}\label{licensing}
 \begin{lstlisting}[language=modelica]
 record Protection
   $\ldots$
-  String features[:] = fill("", 0) "Required license features";
+  /*literal*/ constant String features[:] = fill("", 0) "Required license features";
```

Comment:
I think we use the simpler syntax for annotations:
```suggestion
  /*literal*/ constant String features[:] = {} "Required license features";
```

This should probably be described in the notation.

Reply:
> I think we use the simpler syntax for annotations:
> 
> This should probably be described in the notation.

The `fill("", 0)` was existing, so I don't think that suggestion belongs in this PR.

Additionally I see `{}` as problematic (in general) at least at the moment, since the goal with this PR is to describe the annotations using Modelica syntax and `{}` isn't currently syntactically valid in Modelica - but I think we should discuss that separately and don't have it blocking this PR.

Reply:
Well, it might belong here if we could agree on defining `{}` as a valid default for annotations.

Reply:
To me the main problem is that `{}` isn't syntactically valid.

I don't think that the minor convenience it brings for annotations outweigh the disadvantages of having something that isn't syntactically valid Modelica - especially when the entire goal of this PR is to re-use the Modelica syntax for annotations. People should be familiar with `fill("", 0)` by now.

To me how to best handle that is clearly a different topic, and shouldn't block this one.

Reply:
Now that `{}` was removed from the allowed input, I am putting my vote in favor of `fill("",0)`. In addition to the problem with `{}` being syntactically invalid, `fill("",0)` contains information about the expected type: it should be an array of string that is empty by default.

Reply:
Let's start this over.  Resolving.